### DISCUSSION TITLE
Version v1.020 of the app and version 11 of the firmware have been released

### DISCUSSION THREAD

**Discussion by fhdm-dev** [fhdm-dev/scoppy#75] [Announcements]
This release enables wireless connection between the Scoppy app and the firmware running on a Pico W  (for those who have upgraded to the premium version of the app).

The lastest firmware is available [here](https://oscilloscope.fhdm.xyz/wiki/Installation-&-Getting-Started.html).
To configure the Pico W to connect to your local network, follow the instructions in the [Getting started with Scoppy and the Pico W](https://oscilloscope.fhdm.xyz/wiki/Getting-started-with-the-Pico-W) guide.

**Comment by Jason-BCIT** [comment]
Wow! Very cool... definitely going to have to check that out... and maybe add a battery input to any future versions of front end design so that it can run independently of a USB plug in. (I'm not sure why I didn't think of that when you mentioned you were working on a PicoW version!!!)

I really like how this will provide even greater protection for the Android device against weird over-voltage situations, relative to the wired connection. I mean, the Android device should be pretty safe with the USB connection... but my students always amaze me at what they can accomplish! If they can figure out how to release the magic smoke from their phone using WiFi, though, I'll be VERY impressed!

**Reply by fhdm-dev** [reply]
That would be impressive :smile:
Hopefully the wireless connection method works as well as USB. I haven't had any feedback yet.

**Reply by Jason-BCIT** [reply]
I've managed to find a reasonably local retailer with Pico W's in stock... they're limiting them to one per order, so I've got one on order now. I look forward to giving it a try. I'll definitely post something here when I get a chance to try it out! (My Android app updated automatically... I can see all the options for WiFi now, but just need to get my hands on a Pico W!)

Meanwhile I've been running some tests on my front end design and comparing it to your front-end board... yours is working great! Mine is going along pretty well but I'm getting very noisy results when using my opamp for anything above unity gain. Hmm... hope to get that sorted out on the weekend.

**Reply by Jason-BCIT** [reply]
My Pico W arrived a couple days ago and I finally got a chance to try it out today. Dropped the Wireless Fscope UF2 file onto the PicoW, plugged it into the Fscope board, followed the instructions and it worked like a charm. Wow!

So cool... what is particularly cool is that the USB cable that my students would need (mini to USB-C) for the USB Scoppy costs only a couple dollars less than a Pico W.... and with the wireless mode turned on they no longer need the cable! And they can no longer short out their phone, no matter what they measure! (I do, of course, underestimate students sometimes... but I'm pretty sure that if they can figure out how to toast a phone wirelessly that they have a strong future in electronic espionage, rather than teaching.)

I didn't connect the PicoW to my home wifi network, though... rather just left it in hotspot mode to make it easier to switch between home and work (where the network security makes it tricky to communicate from one device to the other).

Then I plugged the PicoW into the front end design that I've been working on... and it worked perfectly there, too! (Can't say that my design reads quite as cleanly as the Fscope, though... but it works well enough that I think I can figure the rest of it out by playing with a few resistors and capacitors and experimenting with a few different op amps.)

So... wow! Awesome stuff. I didn't do a through testing of both channels, nor the logic analyzer, but the signal generator worked just fine. The wireless features are pretty slick!

When you do have time to post the PicoW code to Github that will be great, as I'm having my students develop custom voltage ranges for their inputs. They can play around with resistor values at both the attenuation stage and the opamp, and then will set the resulting voltage into code and compile it. We're a couple weeks away from that point, though... and even then they can just make temporary changes to the voltage range as an interim solution.

And in that context, while I hate to ask for one more thing... it would be even easier if Scoppy had a way to remember custom voltage ranges from within the app. That way I could just have the students load a common .uf2 onto their Picos, and then tweak the voltage ranges to match their board without having to recompile. If that presents a programming pain, then no worries... you've made it super easy to enter the data into the code, and the ability to make your own board with custom voltage ranges is something that sets Scoppy apart from some of the other USB scopes out there where the software is proprietary and designed to match one specific front end design.

Thank you again for everything you've done on this project... I've had a lot of fun (and a lot of learning) with it. I have definitely discovered a few opamps that didn't behave quite the way I expected them to and this is one of the first board designs that I've made where parasitic capacitance and signal interference has smacked me around a bit!

Jason

**Reply by fhdm-dev** [reply]
Thanks for the feedback. I'll definitely add that change to the app. I've been meaning to do it for while now anyway :smile: 

As for posting the code to Github, I haven't decided yet if I'll open-source the PicoW version of the firmware. If you like I can send you a uf2 with the appropriate voltage ranges for your front end - just send me the configuration settings.

**Reply by Jason-BCIT** [reply]
Cool! If we can lock in custom voltage ranges in a later release of the firmware then we won't need a custom-compiled version of the firmware.

I've taken advantage of the opportunity to have custom voltage ranges to allow the students (cough... cough... REQUIRE the students) to select their own resistor combinations for the input voltage divider and op-amp feedback loops so that they can calculate the resulting attenuation and gain as a learning exercise. I expect everyone will have slightly different voltage ranges on at least a couple of the inputs. I was planning on having them tweak the code and compile it for their board, which is a great excuse to show them how you've set the files up to make this fairly easy to do... but I can still show them that in the USB software, but encourage them to buy a PicoW and use a wireless connection.

In my testing work so far I'll just keep a note of my voltage ranges and punch them in when I start Scoppy. So that will work fine for the students for now, too... and if a variable voltage range version of the PicoW software is in the works then they will be able to upgrade to the latest firmware when you release it. As I mentioned to them  when introducing the project, one of the things that makes Scoppy fun and exciting is that it is under active development and you keep making it better and better!

Thanks again for all your work... it is making for a great project. Once I learn a bit more about how the boards I've designed work, and get some feedback from the students I'll be sharing that design (and the associated learning material) more publicly, too. The students are great at pointing out places where I've missed something or an explanation doesn't make sense!

**Reply by fhdm-dev** [reply]
OK. That's good to hear. I'll try and get that functionality into the app ASAP.

**Reply by fhdm-dev** [reply]
Hi Jason,
The app (version 1.021) now has an option to prevent the voltage ranges being overwritten. It's in the voltage range settings dialog below where you enter the voltage ranges :smiley:

**Reply by Jason-BCIT** [reply]
Thank you so much! I've had five or six students complete my version of a Scoppy Front End so far and it is working out great. Some are using a "W" with the wireless version, others using a cabled version. One student, with an older phone (Android 6.0) couldn't get the Play Store to install the app, but was able to figure out how to sideload it.

I'll let the students know to download the updated app... oh, wait... just checked that and it has auto-updated on my phone already! So I'll let them know the update is available and that they can hard-code their voltage ranges now.

One interesting (and perhaps unavoidable) observation that I've noticed is that when I plug the PicoW into my computer's USB port to provide the power supply it seems to interrupt the process of establishing a local hot spot. I have to plug Scoppy into a 'dumb' USB charger rather than the computer to make a connection. It's an easy thing to do, and if you have already documented that somewhere (and I obviously haven't read it) my apologies! In the next version of my board this will not be an issue as I'll be putting battery inputs onto the board so that it can run off a couple AA cells or something similar.

Thanks again... very much looking forward to seeing what is coming down the pipe in the next six months or so!!

**Reply by fhdm-dev** [reply]
> One student, with an older phone (Android 6.0) couldn't get the Play Store to install the app, but was able to figure out how to sideload it.

Hmmm. My guess is that the phone doesn't support USB OTG. Of course now that's not necessary so I guess I should make the app available to non-OTG phones.

> One interesting (and perhaps unavoidable) observation that I've noticed is that when I plug the PicoW into my computer's USB port to provide the power supply it seems to interrupt the process of establishing a local hot spot.

When you plug the Pico W into a device that has the USB data lines enabled (eg. a computer or an Android device) then the firmware on the Pico W will first try and connect over USB. Then after 10 seconds it will fall back to Wi-Fi mode and only then will it establish the hot spot (access point) which will take a few more seconds. If you're not seeing the access point after about 15 seconds then I think there must be a bug in the firmware.

This is probably something I should fix. It would make more sense for the firmware to listen for connections over both Wi-Fi and USB simultaneously when it first starts up.

> Thanks again... very much looking forward to seeing what is coming down the pipe in the next six months or so!!

Some possibilities (in no particular order) include:
* Math Channel
* Calibration
* Auto-Set
* Revamped UI for tablets
* Statistics
* Aquisition modes eg. Hi-Res
* Configurable pins mappings (without having to re-compile the firmware)
* Protocol decoding (logic analyzer mode)
* More signal generator options
* 9-bit and/or 10-bit sampling
* Sin(x)/x interpolation
* More/better AFE designs
* Support for controlling the scope with mechanical buttons/encoders
* Bluetooth connectivity once that's available on the Pico W
* A new app icon - the current one is actually a microwave oven icon :smiley: 
* Any other feature that is common on oscilloscopes and is practical to implement!

**Reply by gilesp1729** [reply]
Sounds great! I'd like to try and do a Windows backend for the wifi version at some stage (if more about the connection protocol can be disclosed, at your convenience of course). Just because that's my sw development background, and there's a spare Windows machine on my workbench in the garage :-)

**Comment by gilesp1729** [comment]
Is the Pico W firmware up on github? Or is it just the USB version? If the wireless version is made public, I will probably revisit the Windows backend and try to get some reliable communication going.

I'll definitely get an Fscope board to build up your AFE (and put it in a nice 3D printed case). I'm also looking at building a slim probe-sized front end with logic probe indicator LED's in it as well.

**Reply by fhdm-dev** [reply]
It's just the USB version that's on github.

The probe idea sounds interesting. I'd never thought of creating a front end like that. If you end up doing it I'd love to see some photos of it in action :smile:

**Comment by Kerner947** [comment]
Hi. Both the USB and W versions work perfectly for me. In the W version, CH2 was placed on pin A2 (34) and the generator was placed on pin GP16 (21). This is what I experienced. When viewed on a "real" oscilloscope, the signals of both are very nice. On the other hand, the rising edges on the raspberry oscilloscope placed on the test panel are quite ugly. I hope this will change when I build it in a final - shaded - form. Overall, I think it's a great, cheap device. Congratulation!.
